Lib/http/cookiejar.py

```diff
@@ -641,7 +650,8 @@ def eff_request_host(request):
 
     """
     erhn = req_host = request_host(request)
-    if "." not in req_host:
+    if "." not in req_host and '[' not in req_host:
```

Comment:
AFAIR, '[' can only appear in the beginning or the end of the host (without .local). It shouldn't appear in the middle of the netloc.